components/webgpu/lib.rs

```diff
@@ -412,6 +419,46 @@ impl WGPU {
                         )
                     }
                 },
+                WebGPURequest::MapReadAsync(sender, buffer_id, device_id, usage, size) => {
+                    let global = &self.global;
+                    let on_read = move |status: wgpu::resource::BufferMapAsyncStatus,
+                                        ptr: *const u8| {
+                        match status {
+                            wgpu::resource::BufferMapAsyncStatus::Success => {
+                                let mut array_buffer = Vec::with_capacity(size as usize);
+                                unsafe {
+                                    std::ptr::copy(ptr, array_buffer.as_mut_ptr(), size as usize);
+                                    array_buffer.set_len(size as usize);
+                                };
+                                if let Err(e) =
+                                    sender.send(Ok(WebGPUResponse::MapReadAsync(array_buffer)))
```

Comment:
I wonder if servo's IPC allows to put stuff in shared memory without going through the regular `Vec` heap?..

Reply:
We do support shared memory: https://docs.rs/ipc-channel/0.14.0/ipc_channel/ipc/struct.IpcSharedMemory.html

Reply:
Probably worth filing a followup issue about investigating using that.

Reply:
Thank you! I've updated the pull request with using `IpcSharedMemory`.